arraycontext/impl/pytato/__init__.py

```diff
@@ -47,14 +47,39 @@
 from arraycontext.metadata import NameHint
 
 import numpy as np
-from typing import Any, Callable, Union, TYPE_CHECKING, Tuple, Type
-from pytools.tag import ToTagSetConvertible
+from typing import Any, Callable, Union, TYPE_CHECKING, Tuple, Type, FrozenSet
+from pytools.tag import ToTagSetConvertible, normalize_tags, Tag
 import abc
 
 if TYPE_CHECKING:
     import pytato
 
 
+def _preprocess_array_tags(tags: ToTagSetConvertible) -> FrozenSet[Tag]:
+    tags = normalize_tags(tags)
+
+    name_hints = [tag for tag in tags if isinstance(tag, NameHint)]
+    if name_hints:
+        name_hint, = name_hints
```

Comment:
Unhelpful error message if the user passes in >1 `NameHint` tags.

Reply:
Unlikely to happen given that it's a `UniqueTag`.

Reply:
Makes sense!